### PR TITLE
refactor: allow dependency injection for pro

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,5 @@ profile.out
 /dist
 *.test
 tests/__snapshot__
+/vendor/github.com/loft-sh/vcluster-pro/
+/vendor/github.com/FabianKramm/private-test/
```
Comment (ThomasK33) [inline .gitignore:18]: Do we need this, or is it a leftover development snippet?

File: cmd/vcluster/cmd/start.go [hunk 2]
```diff
@@ -66,31 +71,42 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
 	}
 
 	// get current namespace
-	currentNamespace, err := clienthelper.CurrentNamespace()
+	controlPlaneConfig, controlPlaneNamespace, controlPlaneService, workloadConfig, workloadNamespace, workloadService, err := pro.GetRemoteClient(options)
 	if err != nil {
 		return err
 	}
+	options.ServiceName = workloadService
+	_ = os.Setenv("NAMESPACE", workloadNamespace)
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:79]: Shouldn't we handle/exit if this error is non-nil?

File: .gitignore (hunk 1 above)
Comment (FabianKramm) [reply .gitignore:18]: Yeah we don't need that, good catch

File: cmd/vcluster/cmd/start.go (hunk 2 above)
Comment (FabianKramm) [reply cmd/vcluster/cmd/start.go:79]: Can add that yes

File: cmd/vcluster/cmd/start.go [hunk 3]
```diff
@@ -47,10 +46,16 @@ func NewStartCommand() *cobra.Command {
 	}
 
 	options.AddFlags(cmd.Flags(), vClusterOptions)
+	pro.AddProFlags(cmd.Flags(), vClusterOptions)
 	return cmd
 }
 
 func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) error {
+	err := pro.ValidateProOptions(options)
+	if err != nil {
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:55]: [Reduced scope of variable](https://github.com/uber-go/guide/blob/master/style.md#reduce-scope-of-variables)
```suggestion
	if err := pro.ValidateProOptions(options); err != nil {
```

File: cmd/vcluster/cmd/start.go [hunk 4]
```diff
@@ -47,10 +46,16 @@ func NewStartCommand() *cobra.Command {
...
+		return err
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:56]:
```suggestion
		return fmt.Errorf("validate pro options: %w", err)
```

File: cmd/vcluster/cmd/start.go [hunk 5]
```diff
@@ -66,31 +71,45 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
...
 
 	// get current namespace
-	currentNamespace, err := clienthelper.CurrentNamespace()
+	controlPlaneConfig, controlPlaneNamespace, controlPlaneService, workloadConfig, workloadNamespace, workloadService, err := pro.GetRemoteClient(options)
 	if err != nil {
 		return err
 	}
+	options.ServiceName = workloadService
+	err = os.Setenv("NAMESPACE", workloadNamespace)
+	if err != nil {
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:80]: [Reduced variable scope](https://github.com/uber-go/guide/blob/master/style.md#reduce-scope-of-variables)
```suggestion
	if err := os.Setenv("NAMESPACE", workloadNamespace); err != nil {
```

File: cmd/vcluster/cmd/start.go [hunk 6]
```diff
@@ -66,31 +71,45 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
...
-	inClusterConfig.QPS = 40
-	inClusterConfig.Burst = 80
-	inClusterConfig.Timeout = 0
-	inClusterClient, err := kubernetes.NewForConfig(inClusterConfig)
+	// init telemetry
+	telemetry.Collector.Init(controlPlaneConfig, controlPlaneNamespace, options)
+
+	// initialize feature gate from environment
+	err = pro.LicenseInit(ctx, controlPlaneConfig, controlPlaneNamespace, options.ProOptions.ProLicenseSecret)
 	if err != nil {
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:89]:
```suggestion
	if err := pro.LicenseInit(ctx, controlPlaneConfig, controlPlaneNamespace, options.ProOptions.ProLicenseSecret); err != nil {
```

File: cmd/vcluster/cmd/start.go [hunk 7]
```diff
@@ -66,31 +71,45 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
...
 		return err
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:90]:
```suggestion
		return fmt.Errorf("pro license init: %w", err)
```

File: cmd/vcluster/cmd/start.go [hunk 8]
```diff
@@ -102,25 +121,36 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
...
+		controlPlaneClient,
+	)
 	if err != nil {
 		return err
 	}
 
+	// start integrated coredns
+	if controllerCtx.Options.ProOptions.IntegratedCoredns {
+		err = pro.StartIntegratedCoreDNS(controllerCtx)
+		if err != nil {
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:146]:
```suggestion
		if err := pro.StartIntegratedCoreDNS(controllerCtx); err != nil {
```

File: cmd/vcluster/cmd/start.go [hunk 9]
```diff
@@ -102,25 +121,36 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
...
 
 	// start proxy
-	err = setup.StartProxy(controllerCtx)
+	err = setup.StartProxy(
+		controllerCtx,
+		controlPlaneNamespace,
+		controlPlaneService,
+		controlPlaneClient,
+	)
 	if err != nil {
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:139]:
```suggestion
	if err := setup.StartProxy(
		controllerCtx,
		controlPlaneNamespace,
		controlPlaneService,
		controlPlaneClient,
	); err != nil {
```

File: cmd/vcluster/cmd/start.go [hunk 10]
```diff
@@ -102,25 +121,36 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
...
 		return err
```
Comment (ThomasK33) [inline cmd/vcluster/cmd/start.go:140]:
```suggestion
		return fmt.Errorf("setup start proxy: %w", err)
```

**Review by ThomasK33 — COMMENTED** [review]
Some style guide recommendations

**Review by ThomasK33 — APPROVED** [review]

**Pull request merged**